tree/ntuple/v7/inc/ROOT/RDaos.hxx

```diff
@@ -202,34 +243,56 @@ public:
       std::vector<RDaosObject::RAkeyRequest> fDataRequests{};
       std::unordered_map<AttributeKey_t, unsigned> fIndices{};
 
-      void Insert(AttributeKey_t attr, const d_iov_t &iov)
+      void Insert(AttributeKey_t attr, const RDaosIov &iov)
       {
          auto [it, ret] = fIndices.emplace(attr, fDataRequests.size());
          unsigned attrIndex = it->second;
 
          if (attrIndex == fDataRequests.size()) {
-            fDataRequests.emplace_back(attr, std::initializer_list<d_iov_t>{iov});
+            fDataRequests.emplace_back(attr, std::initializer_list<d_iov_t>{iov.Get()});
          } else {
-            fDataRequests[attrIndex].fIovs.emplace_back(iov);
+            fDataRequests[attrIndex].fIovs.emplace_back(iov.Get());
          }
       }
 
-      void Insert(AttributeKey_t attr, std::vector<d_iov_t> &iovs)
+      void Insert(AttributeKey_t attr, RDaosIovV &iovs)
       {
          auto [it, ret] = fIndices.emplace(attr, fDataRequests.size());
          unsigned attrIndex = it->second;
+         auto iovSeq = iovs.Get();
 
          if (attrIndex == fDataRequests.size()) {
-            fDataRequests.emplace_back(attr, iovs);
+            fDataRequests.emplace_back(attr, iovSeq);
          } else {
             fDataRequests[attrIndex].fIovs.insert(std::end(fDataRequests[attrIndex].fIovs),
-                                                  std::make_move_iterator(std::begin(iovs)),
-                                                  std::make_move_iterator(std::end(iovs)));
+                                                  std::make_move_iterator(std::begin(iovSeq)),
+                                                  std::make_move_iterator(std::end(iovSeq)));
          }
       }
    };
 
-   using MultiObjectRWOperation_t = std::unordered_map<ROidDkeyPair, RWOperation, ROidDkeyPair::Hash>;
+   /// \brief Manages request batches for multiple objects. Internally, a dictionary coalesces the requests by their
+   /// co-locality, i.e., their `ROidDkeyPair`. Provides iterator access to the internal data structure.
+   class MultiObjectRWOperation {
+   public:
+      using RequestDict_t =
+         std::unordered_map<RDaosContainer::ROidDkeyPair, RWOperation, RDaosContainer::ROidDkeyPair::Hash>;
+      [[nodiscard]] auto begin() const { return fRequestDict->begin(); }
+      [[nodiscard]] auto end() const { return fRequestDict->end(); }
+      [[nodiscard]] size_t GetSize() const { return fRequestDict->size(); }
+
+      MultiObjectRWOperation() { fRequestDict = std::make_unique<RequestDict_t>(); }
+
+      void Insert(RDaosKey &key, const RDaosIov &pageIov) const
+      {
+         auto odPair = RDaosContainer::ROidDkeyPair(key.fOid.Get(), key.fDkey);
+         auto [it, ret] = fRequestDict->emplace(odPair, RWOperation(odPair));
+         it->second.Insert(key.fAkey, pageIov);
+      }
+
+   private:
+      std::unique_ptr<RequestDict_t> fRequestDict;
```

Comment:
Why not just
```suggestion
      RequestDict_t fRequestDict;
```
?  That should save some unneeded heap allocations/deallocations.